worker/uniter/runner/jujuc/jujuctesting/unit.go

```diff
@@ -6,12 +6,14 @@ package jujuctesting
 import (
 	"github.com/juju/errors"
 	"gopkg.in/juju/charm.v6"
+	"github.com/juju/juju/apiserver/params"
 )
 
 // Unit holds the values for the hook context.
 type Unit struct {
 	Name           string
 	ConfigSettings charm.Settings
+	GoalState      params.StringResult
```

Comment:
goal state should be a string
if we do decide to go with a struct, we will never add a params struct here, we'll use a proper goal state struct